src/openforms/prefill/attributes_generator.py

```diff
@@ -0,0 +1,122 @@
+        if self.url:
+            # The content types supported in the openapi3-parser are these:
+            # https://github.com/manchenkoff/openapi3-parser/tree/master/src/openapi_parser/home/silvia/repositories/openapi3-parser/src/openapi_parser/enumeration.py
+            # However, in the haal centraal OAS we find content types such as "application/hal+json". So with
+            # strict_enum=False, the parser just checks that the content type is a string
+            return parse(uri=self.url, strict_enum=False)
+
+        raise AttributeGeneratorException(
+            "No URL to the OAS specifications was provided."
+        )
+
+    def get_attributes(self, property_schema):
+        attributes = []
+        if hasattr(property_schema.schema, "properties"):
+            for schema_property in property_schema.schema.properties:
+                nested_attributes = self.get_attributes(schema_property)
+                attributes += [
+                    [property_schema.name] + nested_attribute
+                    for nested_attribute in nested_attributes
+                ]
+        else:
+            attributes.append([property_schema.name])
+        return attributes
+
+    def get_schemas_attributes(self, oas_specifications):
+        # In the specifications, the schemas are under the path "components > schemas"
+        parsed_schemas = oas_specifications.schemas
+
+        if self.schema:
+            if self.schema not in parsed_schemas:
+                raise AttributeGeneratorException(
+                    f'The given schema name "{self.schema}" could not be found in the specifications'
+                )
+
+            return [
+                self.get_attributes(schema_property)
+                for schema_property in parsed_schemas[self.schema].properties
+            ]
+
+        raise AttributeGeneratorException("No schema specified")
+
+    def format_attributes(self, attributes):
+        """Turn each attribute in ``{"attribute_variable_name": "bla_bla", "attribute_value": "bla.bla", "attribute_label": "Bla > Bla"}``"""
+        formatted_attributes = []
+        for nested_attributes in attributes:
+            for attribute in nested_attributes:
+                formatted_attributes.append(
+                    {
+                        "attribute_variable_name": (
+                            "_".join(
+                                [attribute_bit.lower() for attribute_bit in attribute]
+                            )
+                        ).lstrip("_"),
+                        "attribute_value": ".".join(
+                            [attribute_bit for attribute_bit in attribute]
+                        ),
+                        "attribute_label": " > ".join(
+                            [attribute_bit.capitalize() for attribute_bit in attribute]
+                        ),
+                    }
+                )
+        return formatted_attributes
+
+    def generate_attributes(self):
```

Comment:
it looks like this is the only public method, so I'd mark the other ones private with an underscore.